sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -257,7 +257,102 @@ class HiveCatalogedDDLSuite extends DDLSuite with TestHiveSingleton with BeforeA
         spark.sql("ALTER TABLE t1 ADD COLUMNS (newcol1 STRUCT<`$col1`:STRING, col2:Int>)")
       }.getMessage
       assert(err.contains("Cannot recognize hive type string:"))
-   }
+    }
+  }
+
+  test("Normal table should inherit database permission") {
```

Comment:
@wangyum . This inherits the parent directory properties and the property name is `DATA_SOURCE_TABLE_INHERIT_PERMS`. Hence, it's not Hive-catalog only stuff.

It would be great if we can have this test coverage on `InMemoryCatalogedDDLSuite`, not only `HiveCatalogedDDLSuite`. We need some refactoring and the following will be the only difference.
```scala
spark.sql("create table datasource_table(a int) using parquet")
spark.sql("create table hive_table(a int)")
```